Product/Production/Services/DocumentQueryCore/src/main/java/gov/hhs/fha/nhinc/docquery/xdsb/helper/XDSbAdhocQueryResponseHelperImpl.java

```diff
@@ -58,12 +58,14 @@ public String getSingleSlotValue(ResponseSlotName slotName, IdentifiableType ide
 
     @Override
     public String getSingleSlotValue(String customSlotName, IdentifiableType identifiableType) {
-        List<SlotType1> slots = identifiableType.getSlot();
-        for (SlotType1 s : slots) {
-            if (StringUtils.equalsIgnoreCase(customSlotName, s.getName()) && s.getValueList() != null) {
-                ValueListType vlt = s.getValueList();
-                for (String string : vlt.getValue()) {
-                    return string;
+        if (identifiableType != null) {
+            List<SlotType1> slots = identifiableType.getSlot();
+            for (SlotType1 s : slots) {
+                if (StringUtils.equalsIgnoreCase(customSlotName, s.getName()) && s.getValueList() != null) {
+                    ValueListType vlt = s.getValueList();
+                    if ((vlt.getValue() != null) && (!vlt.getValue().isEmpty())){
```

Comment:
Unneeded parens around individual statements -- I guarantee that SonarQube will complain about it.